shared/rocroller/test/catch/mxDataGeneratorTest.cpp

```diff
@@ -93,4 +93,148 @@ namespace mxDataGeneratorTest
+                    auto secondGenScale = dgen.getScaleBytes();
+                    auto secondGenRef   = dgen.getReferenceFloat();
+
+                    CHECK(firstGenData[seed] == secondGenData);
+                    CHECK(firstGenScale[seed] == secondGenScale);
+                    CHECK(firstGenRef[seed] == secondGenRef);
+                }
+            }
+
+            omp_set_num_threads(originalThreads);
+        }
+    }
+
+    TEMPLATE_TEST_CASE(
+        "check mxDataGenerator different thread count produces mostly different data",
+        "[mxDataGenerator]",
+        FP4,
+        FP6,
+        BF6,
+        FP8,
+        BF8,
+        Half,
+        BFloat16,
+        float)
+    {
+        SKIP("Current mxdatagenerator does not guarantee same data with different thread "
+             "counts");
+        SUPPORTED_ARCH_SECTION(arch)
+        {
+            const int   dim1         = 32;
+            const int   dim2         = 32;
+            const float min          = -1.f;
+            const float max          = 1.f;
+            const int   blockScaling = 32;
+
+            using rrDT   = TestType;
+            using DGenDT = typename rrDT2DGenDT<rrDT>::type;
+
+            auto             dataType = TypeInfo<rrDT>::Var.dataType;
+            TensorDescriptor desc(dataType, {dim1, dim2}, "T");
+
+            int originalThreads = omp_get_max_threads();
+
+            std::vector<int> threadCounts = {2, 4, 7, 8}; // ref is 1
+            threadCounts.erase(
+                std::remove_if(threadCounts.begin(),
+                               threadCounts.end(),
+                               [originalThreads](int count) { return count > originalThreads; }),
+                threadCounts.end());
+
+            std::vector<uint32_t> seeds = {9861u, 516848u};
+
+            for(uint32_t seed : seeds)
+            {
+                omp_set_num_threads(1);
+                const auto refGen   = getDataGenerator<rrDT>(desc, min, max, seed, blockScaling);
+                auto       refData  = refGen.getDataBytes();
+                auto       refScale = refGen.getScaleBytes();
+                auto       refFloat = refGen.getReferenceFloat();
+
+                unsigned int dataCorrect  = 0;
+                unsigned int scaleCorrect = 0;
+                unsigned int floatCorrect = 0;
+
+                for(int threadCount : threadCounts)
+                {
+                    omp_set_num_threads(threadCount);
+
+                    const auto testGen = getDataGenerator<rrDT>(desc, min, max, seed, blockScaling);
+                    auto       testData  = testGen.getDataBytes();
+                    auto       testScale = testGen.getScaleBytes();
+                    auto       testFloat = testGen.getReferenceFloat();
+
+                    dataCorrect += (refData == testData) ? 1 : 0;
+                    scaleCorrect += (refScale == testScale) ? 1 : 0;
+                    floatCorrect += (refFloat == testFloat) ? 1 : 0;
```

Comment:
nit but up to you whether you want to change it:
```suggestion
                    if (refData == testData)
                        dataCorrect++;
                    if (refScale == testScale)
                        scaleCorrect++;
                    if (refFloat == testFloat)
                        floatCorrect++;
```
or
```suggestion
                    dataCorrect += (refData == testData);
                    scaleCorrect += (refScale == testScale);
                    floatCorrect += (refFloat == testFloat);
```

Otherwise, LGTM.